File: runtime.go
package main

// cloudSyncCapable reports whether THIS runtime owns cloud synchronization
// (R6.0). The CLI Web server is not a cloud-sync product target: all of its
// browser clients already share the server's one canonical data directory, so
// it starts no scheduler and every /api/sync route returns a single stable
// unavailable response. Wails owns the reviewed R0–R5 Go engine and scheduler.
//
// The default follows the build tags: the CLI build (no Wails tag) leaves it
// false (the zero value); the Wails build (production/dev/bindings) sets it
// true in runtime_wails.go. Tests select each runtime mode explicitly by
// assigning this variable, so one test binary can prove the whole matrix
// instead of relying on ambient build-tag globals.
var cloudSyncCapable bool

File: frontend/src/composables/runtime.js
// Runtime capability matrix (R6.0, R6.5). Cloud-sync availability is a property
// of the owning runtime, not just of the frontend build type:
//
//   Wails desktop            -> available (reviewed Go R0–R5 engine + scheduler)
//   CLI Web server (browser) -> unavailable (all clients share one server vault)
//   Pure frontend / PWA      -> available (R6.5 in-page browser engine)
//
// The Wails runtime is detected by the window.go bridge the Wails runtime
// injects. The value is a module-level getter so unit suites can select each
// mode explicitly via setCloudSyncAvailable, mirroring how the Go runtime
// matrix tests override cloudSyncCapable.
export const isLocalBuild = import.meta.env.VITE_LOCAL === '1'
export const isWailsApp = typeof window !== 'undefined' && typeof window.go !== 'undefined'

let _cloudSyncAvailable = isWailsApp || isLocalBuild

export function cloudSyncAvailable() {
  return _cloudSyncAvailable
}

export function setCloudSyncAvailable(value) {
  _cloudSyncAvailable = value
}
